apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -155,8 +164,7 @@ export default class CdoTrashcan extends GoogleBlockly.DeleteArea {
       let trashcanVisibility = 'hidden';
       let toolboxVisibility = 'visible';
       // Don't show the trashcan if the block is being dragged out of the toolbox.
-      const isDraggingFromToolbox =
-        !!Blockly.mainBlockSpace?.currentGesture_?.flyout_;
+      const isDraggingFromToolbox = !!this.workspace.currentGesture_?.flyout_;
```

Comment:
Do we need to keep the `?` after workspace? So `this.workspace?.currentGesture_?.flyout_`? (I assume, in reality, one wouldn't run into issues with this, but maybe these workspaces technically start as undefined?)

Reply:
I think we are good. We always initialize the trashcan [with a workspace](https://github.com/code-dot-org/code-dot-org/blob/655b8f98bebf5a658403452b0d14d9f1175a4a83/apps/src/blockly/googleBlocklyWrapper.js#L663) and we already call things on `this.workspace` in `init`, so if there were an issue with it being null it would have come up.